.azuredevops/pipelines/pr-ci.yml

```diff
@@ -61,32 +79,78 @@ jobs:
       del %TEMP%\vs_buildtools.exe
 
       echo Current MSBuild version:
-      "$(VsInstallDir)\MSBuild\Current\Bin\amd64\MSBuild.exe" --version
+      "$(MSBuildPath)" --version
     displayName: 'Install VS'
 
-  - template: templates\build.yml
+  - template: templates\e2e-test.yml
     parameters:
       RepoRoot: $(Build.SourcesDirectory)
+      MSBuildPath: $(MSBuildPath)
 
-  - template: templates\test.yml
+  - publish: $(LogDirectory)
+    displayName: Publish Logs
+    artifact: '$(System.JobDisplayName) logs'
+    condition: always()
+
+- job: E2ETestVSPreview
+  displayName: E2E Tests using VS Preview
+  dependsOn: Build
+  variables:
+    VsInstallDir: $(Build.ArtifactStagingDirectory)\vs
+    MSBuildPath: $(VsInstallDir)\MSBuild\Current\Bin\amd64\MSBuild.exe
+  steps:
+  - download: current
+    displayName: 'Download Build Artifacts'
+    artifact: artifacts
+
+  - checkout: self
+
+  - task: Cache@2
```

Comment:
Does `vs_buildtools.exe` leverage existing files?

Reply:
Also, will this make us stuck on an old preview version?

Reply:
The cache task just turns the install operation into an update operation. ie we don't avoid the call to the VS installer when the cache task succeeds.